### PR TITLE
:technologist: Update docker environment and optimise for local development

### PR DESCRIPTION
**Pull request by jeyemwey**
As asked in the discord, I simplified the docker compose environment so it is more usable in local development.

This PR does a few things:
* Document how the docker-compose environment works
* Remove _two_ unused/redundant compose files
* Clean up the remaining `docker-compose.yml` to always use the local build and a `.env.docker`
* Hide/Ignore state files from devenv/direnv and docker volumes more strictly

I hope that traewelling and laravel newbies can work with the changes better and contribute to the project quicker.

Meta: There are a bunch of changes in a large number of files, so especially for the `README.md` and the `docker-compose.yml` it would be helpful to just read through the final result.

### CODE REVIEW

File: README.md
Comment (pReya) [inline README.md]: Could you add more info for the seeding variable? Can this stay in the environment permanently? Will the database always be reset & seeded as long as this variable is set?

File: docker-compose.yml
Comment (pReya) [inline docker-compose.yml:86]: I'm using arm architecture, and the docker images for this architecture were only recently added to the db-rest registry. But due to the version pinning I can't use them. Can you explain the reason for pinning the version?

**Review by pReya — CHANGES_REQUESTED** [review]
Thank you very much @jeyemwey! I appreciate all these changes. I have just two questions/concerns.

File: README.md
Comment (jeyemwey) [reply README.md]: It will run the seeder each time you start the container, and add new data every time. There won't be a reset. I'd run it once and then remove it.

File: docker-compose.yml
Comment (jeyemwey) [reply docker-compose.yml:86]: It should be fine as long as we're using v5, the pinning was because I was certain that it worked with that version.

File: README.md
Comment (pReya) [reply README.md]:
```suggestion
You can generate sample data by setting the environment variable `SEED_DB: true` for the `app` container. This will seed the database on restart, when the variable is present.
```

File: docker-compose.yml
Comment (pReya) [inline docker-compose.yml]:
```suggestion
    image: derhuerst/db-rest:5
```

**Review by pReya — CHANGES_REQUESTED** [review]
Without specifying the git hash and just using the `5` version number for `derhuerst/db-rest`, I can finally build/start the project on my M1/ARM Mac. Thanks!

However I'm still getting tons of errors from the seeding process:
```
2023-10-23 18:41:33 Seeding database
2023-10-23 18:41:33 
2023-10-23 18:41:33   Dropping all tables ............................................... 4ms FAIL
2023-10-23 18:41:33 
2023-10-23 18:41:33    UnexpectedValueException 
2023-10-23 18:41:33 
2023-10-23 18:41:33   There is no existing directory at "/usr/src/trwl/storage/logs" and it could not be created: Permission denied
2023-10-23 18:41:33 
2023-10-23 18:41:33   at vendor/monolog/monolog/src/Monolog/Handler/StreamHandler.php:197
2023-10-23 18:41:33     193▕             set_error_handler([$this, 'customErrorHandler']);
2023-10-23 18:41:33     194▕             $status = mkdir($dir, 0777, true);
2023-10-23 18:41:33     195▕             restore_error_handler();
2023-10-23 18:41:33     196▕             if (false === $status && !is_dir($dir) && strpos((string) $this->errorMessage, 'File exists') === false) {
2023-10-23 18:41:33   ➜ 197▕                 throw new \UnexpectedValueException(sprintf('There is no existing directory at "%s" and it could not be created: '.$this->errorMessage, $dir));
2023-10-23 18:41:33     198▕             }
2023-10-23 18:41:33     199▕         }
2023-10-23 18:41:33     200▕         $this->dirCreated = true;
2023-10-23 18:41:33     201▕     }
2023-10-23 18:41:33 
2023-10-23 18:41:33       +10 vendor frames 
2023-10-23 18:41:33 
2023-10-23 18:41:33   11  app/Exceptions/Handler.php:41
2023-10-23 18:41:33       Illuminate\Foundation\Exceptions\Handler::report(Object(Illuminate\Database\QueryException))
2023-10-23 18:41:33       +3 vendor frames 
2023-10-23 18:41:33 
2023-10-23 18:41:33   15  artisan:37
2023-10-23 18:41:33       Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

**Comment by pReya** [comment]
@jeyemwey Do you mind taking a look at the errors from my latest post? Any idea, why the seeding fails due to permission problems?

**Comment by jeyemwey** [comment]
I'm currently on holiday and don't have a development environment with me. Please give me another week or so.

In the mean time, can you try creating the subfolders on the host system? Maybe it's just a permissions thing.

**Comment by jeyemwey** [comment]
I'm currently on holiday and don't have a development environment with me. Please give me another week or so until I can merge the things into `develop`. I don't reckon that other maintainers want to do much of a review here, but your critique has been super helpful.

In the mean time, can you try creating the subfolders on the host system? Maybe it's just a permissions thing.

**Comment by pReya** [comment]
> I'm currently on holiday and don't have a development environment with me. Please give me another week or so until I can merge the things into `develop`. I don't reckon that other maintainers want to do much of a review here, but your critique has been super helpful.
> 
> In the mean time, can you try creating the subfolders on the host system? Maybe it's just a permissions thing.

Could you take another look? The subfolders already exist on the host system

File: docker-compose.yml
Comment (jeyemwey) [reply docker-compose.yml:86]: Resolved with 1cd34720

File: README.md
Comment (jeyemwey) [reply README.md]: Resolved with 1cd34720

**Comment by jeyemwey** [comment]
Hi, sorry for the long wait.

I have pushed 1cd34720 and 4dc0564 which address a number of issues:
* The version tag of `db-rest` has been updated as proposed.
* The seeding now runs in the application directory (`/var/www/html`) and only in the `app` container. This should resolve the `There is no existing directory at "/usr/src/trwl/storage/logs" and it could not be created: Permission denied.` error.
* `SEED_DB` now generates OAuth master keys (and overwrites existing ones on restart). This allows the app to hand out client keys which is necessary for third party apps and various lazy-loaded elements in the frontend (i.e. the notifications).
* Documented what `SEED_DB` does and described that avatars from the `UserFactory` need to be present in the avatars volume mount.

I have now tested (on my M1 mac) like this:

```shell
cd /tmp
git clone git@github.com:Traewelling/traewelling.git
cd traewelling
git checkout dev/jv/docker-env
vi docker-compose.yml # to activate the `SEED_DB` env
docker-compose up --build -d
# wait until the containers are ready
open "http://localhost:8081"
```

Please check and let me know if it works for you :)

**Comment by MrKrisKrisu** [comment]
Hey @jeyemwey and @pReya, can you please have a look on this PR, as this is open for a long time now without any interactions? Thanks!

**Comment by jeyemwey** [comment]
From my side, it's good to go, but I haven't tested on anything but my Mac.

**Comment by MrKrisKrisu** [comment]
> From my side, it's good to go, but I haven't tested on anything but my Mac.

Then I would suggest that we merging this soon if nobody puts a veto. :)

**Pull request merged**
